ironfish-cli/src/commands/wallet/prune.ts

```diff
@@ -27,10 +31,52 @@ export default class PruneCommand extends IronfishCommand {
     CliUx.ux.action.start(`Opening node`)
     const node = await this.sdk.node()
     await NodeUtils.waitForOpen(node)
-    await node.wallet.open()
-    await node.wallet.walletDb.open()
     CliUx.ux.action.stop('Done.')
 
+    if (!node.chain.synced) {
+      this.log(
+        `Your node must be synced with the Iron Fish network to prune wallet transactions. Please try again later`,
+      )
+      this.exit(1)
+    }
+
+    const walletHeadHash = await node.wallet.getLatestHeadHash()
+
+    if (walletHeadHash === null) {
+      this.log(`Failed to get latest head hash.`)
+      this.exit(1)
+      return
+    }
+
+    const head = await node.chain.getHeader(walletHeadHash)
+
+    if (head === null) {
+      this.log(`Failed to get chain header.`)
+      this.exit(1)
+      return
+    }
+
+    for (const account of node.wallet.listAccounts()) {
+      let count = 0
+
+      for await (const { transaction } of account.getExpiredTransactions(head.sequence)) {
```

Comment:
this should use the head of each sequence, so account.getHead() and use the sequence from there. the reason is we don't want to delete things that are expired on the chain if we haven't processed them yet by the wallet. It's possible it could be confirmed if the wallet is just behind the chain in processing the wallet.

Reply:
updated